ethereum/spec/src/main/java/tech/pegasys/teku/spec/SpecProvider.java

```diff
@@ -81,7 +85,39 @@ public Bytes4 domainBeaconProposer(final UInt64 epoch) {
     return atEpoch(epoch).getConstants().getDomainBeaconProposer();
   }
 
+  public UInt64 computeEpochAtSlot(final UInt64 slot) {
+    return atSlot(slot).getBeaconStateUtil().computeEpochAtSlot(slot);
+  }
+
   public Fork fork(final UInt64 epoch) {
     return forkManifest.get(epoch);
   }
+
+  // Validator Utils
```

Comment:
Open to discussing whether exposing spec logic directly from `SpecProvider` is a good pattern to adopt

Reply:
i really think this simplifies our life , i'd prefer that we have these at least for where its a heavily used constant or function...